include/devicetree/dma.h

```diff
@@ -0,0 +1,242 @@
+ *
+ *     dma1: dma@... {
+ *             label = "DMA_1";
+ *     };
+ *
+ *     dma2: dma@... {
+ *             label = "DMA_2";
+ *     };
+ *
+ *     n: node {
+ *		dmas = <&dma1 1 0x400>,
+ *		       <&dma2 6 0x404>;
+ *		dma-names = "tx", "rx";
+ *     };
+ *
+ * Bindings fragment for the dma nodes:
+ *
+ *     dma-cells:
+ *       - channel
+ *       - config
+ *
+ * Example usage:
+ *
+ *     DT_DMAS_CELLS_BY_IDX(DT_NODELABEL(n), channel, 0) // 1
+ *     DT_DMAS_CELLS_BY_IDX(DT_NODELABEL(n), config, 1) // 0x404
+ *
+ * @param node_id node identifier
+ * @param cell_name binding's cell name within the specifier at index idx
+ * @param idx logical idx of the dma channel specifier
+ * @return the value of "cell" inside the specifier
+ */
+
+#define DT_DMAS_CELLS_BY_IDX(node_id, cell_name, idx) \
+	DT_PHA_BY_IDX(node_id, dmas, cell_name, idx)
+
+/**
+ * @brief Get dma channel specifier "cell" value for a given cell idx
+ * (see @ref DT_DMAS_CELLS_BY_IDX)
+ * @param inst instance number
+ * @param cell_name binding's cell name within the specifier at index idx
+ * @param idx logical idx of the dma channel specifier
+ * @return the value of "cell" inside the specifier
+ */
+#define DT_INST_DMAS_CELLS_BY_IDX(node_id, cell_name, idx) \
+	DT_PHA_BY_IDX(DT_DRV_INST(inst), dmas, cell_name, idx)
+
+/**
+ * @brief Get dma channel specifier "cell" value for a given cell name
+ *
+ * Example devicetree fragment:
+ *
+ *     dma1: dma@... {
+ *             label = "DMA_1";
+ *     };
+ *
+ *     dma2: dma@... {
+ *             label = "DMA_2";
+ *     };
+ *
+ *     n: node {
+ *		dmas = <&dma1 1 0x400>,
+ *		       <&dma2 6 0x404>;
+ *		dma-names = "tx", "rx";
+ *     };
+ *
+ * Bindings fragment for the dma nodes:
+ *
+ *     dma-cells:
+ *       - channel
+ *       - config
+ *
+ * Example usage:
+ *
+ *     DT_DMAS_CELLS_BY_NAME(DT_NODELABEL(n), channel, tx) // 1
+ *     DT_DMAS_CELLS_BY_NAME(DT_NODELABEL(n), config, rx) // 0x404
+ *
+ * @param node_id node identifier
+ * @param cell_name binding's cell name within the specifier referenced
+ * as "name"
+ * @param name reference of the tatgeted dma channel specifier
+ * @return the value of "cell" inside the specifier
+ * @see DT_PHA_PHANDLE_IDX()
+ */
+
+#define DT_DMAS_CELLS_BY_NAME(node_id, cell_name, name) \
+	DT_PHA_BY_NAME(node_id, dmas, cell_name, name)
+
+/**
+ * @brief Get dma channel specifier "cell" value for a given cell name
+ * (see @ref DT_DMAS_CELLS_BY_NAME)
+ * @param inst instance number
+ * @param cell_name binding's cell name within the specifier referenced
+ * as "name"
+ * @param name reference of the tatgeted dma channel specifier
+ * @return the value of "cell" inside the specifier
+ */
+#define DT_INST_DMAS_CELLS_BY_NAME(inst, cell_name, name) \
+	DT_DMAS_CELLS_BY_NAME(DT_DRV_INST(inst), cell_name, name)
+
+/**
+ * @brief Does a DMA client has a DMA specifer referenced as "name"
+ * @param node_id node identifier
+ * @param name dma channel specifier name
```

Comment:
When you get a chance to update the docs, I usually put "lowercase-and-underscores" for any parameters that have to be converted that way.

Reply:
Done